tests/test_install_mender_sh.py

```diff
@@ -225,25 +225,114 @@ def test_configure(
         check_installed(generic_debian_container, "mender-connect", installed=False)
         check_installed(generic_debian_container, "mender-configure")
 
-    def test_upgrade_all_packages(
+    def test_upgrade_mender_v3_series(
         self, generic_debian_container,
     ):
-        # Install latest stable software
+        # Install latest stable software (client v3)
         generic_debian_container.run(
             f"curl http://{SCRIPT_SERVER_ADDR}:{SCRIPT_SERVER_PORT}/install-mender.sh | bash -s"
         )
 
-        # And now upgrade to the freshly built packages
+        # Now upgrade to the freshly built packages
         local_apt_repo_from_built_packages(generic_debian_container)
-        generic_debian_container.run("apt -y upgrade")
+        generic_debian_container.run("apt --assume-yes upgrade")
 
-        # All packages should be upgraded
+        # Only mender-client should be upgraded
         check_installed(generic_debian_container, "mender-client")
+        check_installed(generic_debian_container, "mender-client4", installed=False)
+        check_installed(generic_debian_container, "mender-update", installed=False)
+        check_installed(generic_debian_container, "mender-auth", installed=False)
+        check_installed(generic_debian_container, "mender-flash", installed=False)
+        check_installed(generic_debian_container, "mender-setup", installed=False)
+        check_installed(generic_debian_container, "mender-snapshot", installed=False)
+        # The addons should not be removed
+        check_installed(generic_debian_container, "mender-connect")
+        check_installed(generic_debian_container, "mender-configure")
+
+    def test_upgrade_mender_v4_series_meta_package_with_addons(
+        self, generic_debian_container,
+    ):
+        # Install latest stable software (client v3)
+        generic_debian_container.run(
+            f"curl http://{SCRIPT_SERVER_ADDR}:{SCRIPT_SERVER_PORT}/install-mender.sh | bash -s"
+        )
+
+        local_apt_repo_from_built_packages(generic_debian_container)
+
+        # MEN-7010
+        # Currently, production mender-configure still depends (only) on mender-client.
+        # We need first to update this one to the freshly built in order to make the rest of the
+        # test succeed. This part can be removed after MEN-7010 is finished.
+        generic_debian_container.run("apt install --assume-yes mender-configure")
+
+        # Now install freshly built mender-client4
+        generic_debian_container.run(
+            "DEBIAN_FRONTEND=noninteractive apt install --assume-yes mender-update mender-client4"
```

Comment:
You specified `mender-update` here. Intentional? Or should it be only `mender-client4`, to improve user experience?

I see you've done it differently below, but I don't quite see the intention behind it.

Reply:
This also was a painful discovery. If you have installed a package that depends on `mender-client | something`. That _something_ needs to be installed explicitly (or with a direct dependency, like `mender-update` depends on `mender-auth`) for the package to be kept in the system. In other words if you install only `mender-client4` apt will refuse to install it unless you remove `mender-connect` or other addons depending on `mender-client` (which has to be removed if you install `mender-client4`).

This is why the below test (without addons) does not require the `mender-update` explicit installation.

Reply:
Ok, this is hard to wrap my head around. Let me try to recap what you said:

* Addons installed
    * `apt install mender-client4` will not work.
    * `apt install mender-client4 mender-update` will work.
* Addons not installed
    * `apt install mender-client4` will work.
    * `apt install mender-client4 mender-update` will work.

Bah! It's so annoying that the *addons* are what is breaking this... But I understand that we've run out of options here.

Reply:
That is correct. For reference, see this extract from the log from `apt install` in the addons case:
```
...
dpkg: mender-client: dependency problems, but removing anyway as you requested:
 mender-connect depends on mender-client (>= 3.2.0) | mender-auth; however:
  Package mender-client is to be removed.
  Package mender-auth is not installed.
  Package mender-client which provides mender-auth is to be removed.
 mender-configure depends on mender-update | mender-client (>= 2.5.0); however:
  Package mender-update is not installed.
  Package mender-client which provides mender-update is to be removed.
  Package mender-client is to be removed.
 mender-connect depends on mender-client (>= 3.2.0) | mender-auth; however:
  Package mender-client is to be removed.
  Package mender-auth is not installed.
  Package mender-client which provides mender-auth is to be removed.
 mender-configure depends on mender-update | mender-client (>= 2.5.0); however:
  Package mender-update is not installed.
  Package mender-client which provides mender-update is to be removed.
  Package mender-client is to be removed.
...
```

Reply:
Got it. We'll have to live with this then. As long as subsequent upgrades will work, we're not too bad off.